<launch>

  <!-- LAUNCH RVIZ -->

  <arg name="debug" default="false" />
  <arg unless="$(arg debug)" name="launch_prefix" value="" />
  <arg     if="$(arg debug)" name="launch_prefix" value="gdb --ex run --args" />

  <arg name="config" default="true" />
  <arg unless="$(arg config)" name="command_args" default="" />
  <arg     if="$(arg config)" name="command_args" default="-d $(find video_survey)/launch/visualizeit.rviz" />

  <include file="$(find moveit_recorder)/launch/planning_context.launch">
    <arg name="load_robot_description" value="true"/>
  </include>

  <node pkg="tf" type="static_transform_publisher" name="virtual_joint_broadcaster_0" args="0 0 0 0 0 0 odom_combined base_footprint 100" />

  <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
    <param name="/use_gui" value="false"/>
    <rosparam param="/source_list">[/move_group/fake_controller_joint_states]</rosparam>
  </node>

  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="false" output="screen" />

  <node name="$(anon rviz)" launch-prefix="$(arg launch_prefix)" pkg="rviz" type="rviz" respawn="false"
      args="$(arg command_args)" output="screen">
    <rosparam command="load" file="$(find pr2_moveit_config)/config/kinematics.yaml"/>
  </node>

  <!-- LAUNCH vis NODE -->

  <arg name="planning_scene_topic" default="planning_scene" />
  <arg name="display_pose_topic" default="visualize_state" />
  <arg name="camera_topic" default="/rviz/camera_placement" />
  <arg name="vis_dir" />
  <arg name="path_dir" />
  <arg name="save_dir" />
  <arg name="vis_traj" default="true" />

  <node name="visualize" pkg="video_survey" type="visualize" 
    args="--planning_scene_topic $(arg planning_scene_topic)
          --display_pose_topic $(arg display_pose_topic)
          --camera_topic $(arg camera_topic)
          --vis_traj $(arg vis_traj)
          --vis_dir $(arg vis_dir)
          --path_dir $(arg path_dir)
          --save_dir $(arg save_dir)" launch-prefix="$(arg launch_prefix)" respawn="false" output="screen">
  </node>

</launch>
